R/utilities-sensitivity-analysis.R

```diff
@@ -1,342 +1,343 @@
-                           resultsFileFolder,
-                           resultsFileName,
-                           popSAResultsIndexFile) {
-  allPKResultsDataframe <- getPKResultsDataFrame(pkParameterResultsFilePath)
-  outputs <- levels(allPKResultsDataframe$QuantityPath)
-  pkParameters <- levels(allPKResultsDataframe$Parameter)
-  # pkParameters<- pkParameters[1:3]#### ONLY TO TEST SMALL PK PARAMETER SET - REMOVE THIS LINE EVENTUALLY!!!!!!!!!!!!!!!!!!!!
-
-
-  outputColumn <- NULL
-  pkParameterColumn <- NULL
-  individualIdColumn <- NULL
-  valuesColumn <- NULL
-  unitsColumn <- NULL
-  quantileColumn <- NULL
-
-  for (output in outputs) {
-    for (pkParameter in pkParameters) {
-      singleOuputSinglePKDataframe <- allPKResultsDataframe[ allPKResultsDataframe["QuantityPath"] == output & allPKResultsDataframe["Parameter"] == pkParameter, ]
-      quantileResults <- getQuantileIndividualIds(singleOuputSinglePKDataframe, quantileVec)
-      for (n in 1:length(quantileResults$ids)) {
-        outputColumn <- c(outputColumn, output)
-        pkParameterColumn <- c(pkParameterColumn, pkParameter)
-        individualIdColumn <- c(individualIdColumn, quantileResults$ids[n])
-        valuesColumn <- c(valuesColumn, quantileResults$values[n])
-        unitsColumn <- c(unitsColumn, quantileResults$units[n])
-        quantileColumn <- c(quantileColumn, quantileVec[n])
-      }
-    }
-  }
-
-  filenamesColumn <-  paste0(sapply(X = individualIdColumn,FUN = getIndividualSAResultsFileName,resultsFileName),".csv")
-  sensitivityAnalysesResultsIndexFileDF <- data.frame("Outputs" = outputColumn, "pkParameters" = pkParameterColumn, "Quantile" = quantileColumn, "Value" = valuesColumn, "Unit" = unitsColumn,  "IndividualId" = individualIdColumn, "Filename" = filenamesColumn)
-  write.csv(x = sensitivityAnalysesResultsIndexFileDF, file = file.path(resultsFileFolder, paste0(popSAResultsIndexFile, ".csv")))
-  return(sensitivityAnalysesResultsIndexFileDF)
-}
-
-
-#' @title getIndividualSAResultsFileName
-#' @description Function to build name of inidividual SA results file
-#' @param resultsFileName root name of population sensitivity analysis results CSV files
-#' @param individualId id of individual
-getIndividualSAResultsFileName <- function(individualId,resultsFileName){
-  return(paste(resultsFileName,"IndividualId",individualId,sep="-"))
-}
+#' @title analyzeSensitivity
+#' @description Determine whether to run SA for individual or population.  If for individual,  pass simulation to individualSensitivityAnalysis.
+#' If SA is for population, loop thru population file, extract parameters for each individual, and pass them to individualSensitivityAnalysis.
+#' @param simFilePath path to simulation file
+#' @param parametersToPerturb paths to parameters to perturb in sensitivity analysis
+#' @param popFilePath path to the population data file
+#' @param individualID ID of individual in population data file for whom to perform sensitivity analysis
```

Comment:
This looks like this is only one individual. But this can be an array right? Maybe it should be individualIDs?
Also what do we use as nomenclature in core ?ID or Id? Let's make sure we are consistent (I can't remember)

Reply:
It is only for one individual.  runSensitivity loops through individual id arrays as need and passes the id number to this function.  

renamed to individualId